src/state-card-custom-ui.html

```diff
@@ -30,36 +30,76 @@
     'inputChanged(hass, inDialog, stateObj)',
   ],
 
-  badgeMode: function (hass, stateObj) {
+  badgeMode: function (hass, stateObj, badgeLabelBorder, badgeLabelText, badgeBackground) {
     var states = [];
-    stateObj.attributes.entity_id.forEach((id) => {
-      var state = hass.states[id];
-      if (!state) {
-        /* eslint-disable no-console */
-        console.warn('Unknown ID ' + id + ' in group ' + stateObj.entity_id);
-        /* eslint-enable no-console */
-        return;
-      }
-      if (!stateObj.attributes.badges_list ||
-          stateObj.attributes.badges_list.indexOf(state.entity_id) >= 0) {
-        states.push(window.customUI.maybeChangeObject(
-          this, state, false /* inDialog */, false /* allowHidden */));
-      }
-    });
+    if ( typeof stateObj.attributes.entity_id !== 'undefined' && window.hassUtil.computeDomain(stateObj) === 'group' ) {
+      stateObj.attributes.entity_id.forEach((id) => {
+        var state = hass.states[id];
+        if (!state) {
+          /* eslint-disable no-console */
+          console.warn('Unknown ID ' + id + ' in group ' + stateObj.entity_id);
+          /* eslint-enable no-console */
+          return;
+        }
+        if (!stateObj.attributes.badges_list ||
+            stateObj.attributes.badges_list.indexOf(state.entity_id) >= 0) {
+          states.push(window.customUI.maybeChangeObject(
+            this, state, false /* inDialog */, false /* allowHidden */));
+        }
+      });
+    } else {
+        var state = hass.states[stateObj.entity_id];
+        if (!state) {
+          /* eslint-disable no-console */
+          console.warn('Unknown ID ' + stateObj.entity_id + ' in group ' + stateObj.entity_id);
+          /* eslint-enable no-console */
+          return;
+        }
+        if (!stateObj.attributes.badges_list ||
+            stateObj.attributes.badges_list.indexOf(state.entity_id) >= 0) {
+          states.push(window.customUI.maybeChangeObject(
+            this, state, false /* inDialog */, false /* allowHidden */));
+        }
+    }
     window.hassUtil.dynamicContentUpdater(
       this,
       'HA-BADGES-CARD',
       { hass: hass, states: states });
     Polymer.dom(this).lastChild.style.setProperty('--ha-state-label-badge-margin-bottom', '0');
+    Polymer.dom(this).lastChild.style.setProperty('--label-badge-red', badgeLabelBorder);
+    Polymer.dom(this).lastChild.style.setProperty('--label-badge-text-color', badgeLabelText);
+    Polymer.dom(this).lastChild.style.setProperty('--label-badge-background-color', badgeBackground);
+    if (Polymer.dom(this).parentNode.parentNode.tagName === "DIV" && Polymer.dom(this).parentNode.parentNode.classList.contains('state')) {
```

Comment:
This CSS will remain even if the element is used later in non-badge mode.

Reply:
Ok, similar to the other one (background-color). Not yet resolved.